lib/environment_file.rb

```diff
@@ -0,0 +1,72 @@
+class EnvironmentFile
+  OPTIONS = %w{
+    graph
+    network_watcher
+  }.freeze
+
+  def initialize(path)
+    @file = File.new(path)
+  end
+
+  def current_options
+    OPTIONS.map do |option|
+      option if key?(option)
+    end.compact
+  end
+
+  def synchronize(envs)
+    to_delete = OPTIONS - envs
+    unknown = envs - OPTIONS
+
+    raise "The following options are not known #{unknown}. Please chose one of #{OPTIONS}." unless unknown.empty?
+
+    (envs - unknown).each { |key| set_var(key.upcase) }
+    to_delete.each { |key| remove_var(key.upcase) }
+  end
+
+  private
+
+  def set_var(key, value = 'true')
+    found, new_content = replace_key(key) do |k, v|
+      "export #{k}=#{v}\n"
+    end
+
+    new_content << "export #{key}=#{value}\n" unless found
+
+    File.write(@file.path, new_content)
+  end
+
+  def remove_var(key)
+    _, new_content = replace_key(key) do |_key, _value|
+      ''
+    end
+
+    File.write(@file.path, new_content)
+  end
+
+  def key?(key)
+    File.open(@file.path, 'r').each_line do |line|
+      if /export #{key.upcase}=.*$/.match?(line)
+        return true
+      end
+    end
+
+    false
+  end
+
+  def replace_key(key, value = nil)
+    new_content = ''
+    found = false
+
+    File.open(@file.path, 'r').each_line do |line|
+      if /export #{key}=.*$/.match?(line)
+        new_content << yield(key, value)
```

Comment:
Rubocop! 
```
                        ^
lib/environment_file.rb:79:24: C: Performance/RedundantBlockCall: Use yield instead of formatter
.call.
        new_content << formatter.call(key, value)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
```